go-controller/pkg/cluster/master.go

```diff
@@ -0,0 +1,143 @@
+package cluster
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+
+	"github.com/golang/glog"
+
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+	kapi "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/origin/pkg/util/netutils"
+)
+
+func (cluster *OvnClusterController) StartClusterMaster(masterNodeName string) error {
+	clusterNetwork := cluster.ClusterIPNet
+	hostSubnetLength := cluster.HostSubnetLength
+
+	subrange := make([]string, 0)
+	existingNodes, err := cluster.Kube.GetNodes()
+	if err != nil {
+		glog.Errorf("Error in initializing/fetching subnets: %v", err)
+		return err
+	}
+	for _, node := range existingNodes.Items {
+		hostsubnet, ok := node.Annotations[OVN_HOST_SUBNET]
+		if ok {
+			subrange = append(subrange, hostsubnet)
+		}
+	}
+	masterSwitchNetwork, err := calculateMasterSwitchNetwork(clusterNetwork.String(), hostSubnetLength)
```

Comment:
It is not clear to me how this will work. Above, we look at all the nodes that have already been allocated a subnet. We call a subnet allocator for master. How do we know that they will not overlap?